src/App.js

```diff
@@ -1,24 +1,76 @@
-import logo from './logo.svg';
-import './App.css';
 
+import './App.css';
+import Die from './components/Die';
+import { useEffect, useState } from 'react';
+import { nanoid } from "nanoid";
+import Confetti from 'react-confetti'
 function App() {
+  const generateDice = () => {
+    return {
+      value: Math.floor((Math.random() * 6) + 1),
+      isHeld: false,
+      id: nanoid()
+    }
+  }
+  const allNewDice = () => {
+    const newDice = []
+    for (let i = 0; i < 10; i++) {
+      newDice.push(generateDice())
+
+    }
+    return newDice
+
+  }
+  const rollDice = () => {
+    if (!tenzies) {
+      setDice(oldDice => oldDice.map(die => {
+        return die.isHeld ?
+          die :
+          generateDice()
+      }))
+    } else {
+      setTenzies(false)
+      setDice(allNewDice())
+    }
+
+  }
+  const HoldDice = (id) => {
+    setDice(oldDice => oldDice.map(die => {
+      return die.id === id ? {
+        ...die, isHeld: !die.isHeld
+      } : die
+    }))
+
+  }
+
+  const [dice, setDice] = useState(allNewDice())
+  const [tenzies, setTenzies] = useState(false)
+  const diceElements = dice.map(die => <Die key={die.id} value={die.value} onClick={() => HoldDice(die.id)} isHeld={die.isHeld} />)
+
+
+  useEffect(() => {
+    const allHeld = dice.every(die => die.isHeld)
+    const firstValue = dice[0].value
+    const allSameValue = dice.every(die => die.value === firstValue)
+    if (allHeld && allSameValue) {
+      setTenzies(true)
+      console.log("You won")
+    }
+
+  }, [dice])
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+
+    <main className='flex flex-col items-center'>
+      <h1 className="title">Tenzies</h1>
+      <p className="instructions">Roll until all dice are the same. Click each die to freeze it at its current value between rolls.</p>
+      <div className=' grid grid-rows-2 grid-cols-5 gap-2 px-9- py-36'>
```

Comment:
you can remove the spaces and remove invalid styles
```suggestion
      <div className='grid grid-rows-2 grid-cols-5 gap-2 py-36'>
```